### PR TITLE
Feature GB type

### LINKED ISSUE #21704: Distinguish High Angle and Low Angle GB and Assign different diffusivities or other grain parameters
## Reason

The element diffusion will be affected by Grain Boundary (GB) type: It diffuse faster in High Angle GB (HAGB) than in Low Angle GB (LAGB). The distinguishing of the two GB types can increase the diffusion model accuracy.

## Design

The feature include two phases. In the first phase, a grain growth simulation will distinguish the GB type based on the misorientation from EBSD data and smooth the sharp GBs. As the following figure shown, LAGB is assign "1", HAGB is assign "2", and the Triple junction is assigned an interpolated value. In the second phase, a diffusion simulation will build the "bnds" for different GB types from the result of first phase. Then, we can use the different "bnds" to assign the continuous local diffusivity.
image

## Impact

With the new feature, MOOSE can read the EBSD data and derive the local diffusivity or other grain parameters. Also, the more accurate effective diffusivity from the feature can be used to evaluate the influence of grain structure, like different grain size, aspect ratio, and the percentage of HAGB.

### PR DESCRIPTION
**Pull request by Linwitness**
#closes #21704

## Reason

The element diffusion will be affected by Grain Boundary (GB) type: It diffuse faster in High Angle GB (HAGB) than in Low Angle GB (LAGB). The distinguishing of the two GB types can increase the diffusion model accuracy.

## Design

The feature include two phases. In the first phase, a grain growth simulation will distinguish the GB type based on the misorientation from EBSD data and smooth the sharp GBs. As the following figure shown, LAGB is assign "1", HAGB is assign "2", and the Triple junction is assigned an interpolated value. In the second phase, a diffusion simulation will build the "bnds" for different GB types from the result of first phase. Then, we can use the different "bnds" to assign the continuous local diffusivity.
image

## Impact

With the new feature, MOOSE can read the EBSD data and derive the local diffusivity or other grain parameters. Also, the more accurate effective diffusivity from the feature can be used to evaluate the influence of grain structure, like different grain size, aspect ratio, and the percentage of HAGB.

### CODE REVIEW

File: modules/phase_field/doc/content/source/auxkernels/SolutionAuxMisoBnds.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# SolutionAuxMisoBnds
```
Comment (dschwen) [inline modules/phase_field/doc/content/source/auxkernels/SolutionAuxMisoBnds.md:1]: Is "Miso" related to the soup?

File: modules/phase_field/include/auxkernels/SolutionAuxMisoBnds.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+class SolutionUserObject;
+
+/**
+ * AuxKernel for reading a solution from file.
+ * Creates a function that extracts values from a solution read from a file,
+ * via a SolutionUserObject. It is possible to scale and add a constant to the
+ * solution read.
+ */
+class SolutionAuxMisoBnds : public AuxKernel
```
Comment (dschwen) [inline modules/phase_field/include/auxkernels/SolutionAuxMisoBnds.h:22]: This is largely a copy paste of `SolutionAux`. You should inherit from that class instead. Also please do not use abbreviations like "Miso" for Misorientation. Just spell it out (same with Bnds -> Boundaries).

Comment (Linwitness) [reply modules/phase_field/include/auxkernels/SolutionAuxMisoBnds.h]: Thank you! I just updated them. Just let me know if you have any suggestions.

**Comment by dschwen** [comment]
Why did you close this PR?

**Comment by Linwitness** [comment]
Because the one following push generates an issue and I cannot rebase it successfully. I opened a new PR with clean push. It has the same name.


From: Daniel Schwen ***@***.***>
Reply-To: idaholab/moose ***@***.***>
Date: Tuesday, August 2, 2022 at 8:11 PM
To: idaholab/moose ***@***.***>
Cc: Linwitness ***@***.***>, State change ***@***.***>
Subject: Re: [idaholab/moose] Feature GB type (PR #21741)


Why did you close this PR?

**Pull request closed without merging**